apps/nuxt3-ssr/components/HorizontalScrollHelper.vue

```diff
@@ -0,0 +1,61 @@
+<script setup>
+import { computed, ref } from "vue";
+
+const props = defineProps({
+  addFade: {
+    type: Boolean,
+  },
+  addScrollButton: {
+    type: Boolean,
+  },
+});
+
+let scrollAtStart = ref(true);
+let scrollAtEnd = ref(false);
+let scrollElement = ref(null);
+
+function scroll(event) {
+  scrollAtStart.value = event.target.scrollLeft === 0;
+  scrollAtEnd.value =
+    event.target.scrollLeft + event.target.offsetWidth ===
+    event.target.scrollWidth;
+}
+
+function smoothScroll(value) {
+  scrollElement.value.scrollTo({
+    left: scrollElement.value.scrollLeft - value,
+    behavior: "smooth",
```

Comment:
nice :) , did you take a look at: https://tailwindcss.com/docs/scroll-snap-type ?